core_java/jdbc/src/main/java/ca/jrvs/apps/jdbc/CustomerDAO.java

```diff
@@ -0,0 +1,108 @@
+package ca.jrvs.apps.jdbc;
+
+import ca.jrvs.apps.jdbc.util.DataAccessObject;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+public class CustomerDAO extends DataAccessObject<Customer> {
+    private static final String INSERT = "INSERT INTO customer (first_name, last_name, " +
+            "email, phone, address, city, state, zipCode) VALUES (?,?,?,?,?,?,?,?)";
+
+    private static final String GET_ONE = "SELECT customer_id, first_name, last_name, "
+            + "email, phone, address, city, state, zipcode FROM customer WHERE customer_id=?";
+
+    private static final String UPDATE = "UPDATE customer SET first_name = ?, last_name = ?, " +
+            "email = ?, phone = ?, address = ?, city = ?, state = ?, zipcode = ? WHERE customer_id = ?";
+
+    private static final String DELETE = "DELETE FROM customer WHERE customer_id=?";
+    public CustomerDAO(Connection connection) {
+        super(connection);
+    }
+
+    @Override
+    public Customer findById(long id) {
+        Customer customer = new Customer();
+        try(PreparedStatement statement = this.connection.prepareStatement(GET_ONE)) {
+            statement.setLong(1,id);
+            ResultSet rs = statement.executeQuery();
+            while(rs.next()) {
+                customer.setId(rs.getLong("customer_id"));
+                customer.setFirstName(rs.getString("first_name"));
+                customer.setLastName(rs.getString("last_name"));
+                customer.setPhone(rs.getString("phone"));
+                customer.setEmail(rs.getString("email"));
+                customer.setState(rs.getString("state"));
+                customer.setZipCode(rs.getString("zipcode"));
+                customer.setCity(rs.getString("city"));
+            }
+
+        }catch(SQLException e) {
+            e.printStackTrace();
```

Comment:
do no printstackrace..use logger instead (see Java coding style Notion page)
fix this for all occurrences